src/datatypes2/src/types/date_type.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use arrow::datatypes::{DataType as ArrowDataType, Date32Type};
+use common_time::Date;
+use serde::{Deserialize, Serialize};
+use snafu::OptionExt;
+
+use crate::data_type::{ConcreteDataType, DataType};
+use crate::error::{self, Result};
+use crate::scalars::ScalarVectorBuilder;
+use crate::type_id::LogicalTypeId;
+use crate::types::LogicalPrimitiveType;
+use crate::value::{Value, ValueRef};
+use crate::vectors::{DateVector, DateVectorBuilder, MutableVector, Vector};
+
+/// Data type for Date (YYYY-MM-DD).
+#[derive(Debug, Default, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DateType {}
```

Comment:
Can we use `pub struct DateType;` ?

Reply:
Oh, maybe `struct DateType;` is better as the [style guide](https://pingcap.github.io/style-guide/rust/data.html) mentions that
> Empty structs should be defined as struct Foo;, not struct Foo {} or struct Foo(); (the latter two should only be used if necessary in macros).